### PR TITLE
feat(console): configuration options per selected entrypoint

### PR DESCRIPTION
**Pull request by gcusnieux**
## Issue

https://gravitee.atlassian.net/browse/APIM-451

## Description

A small description of what you did in that PR.

## Additional context

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-ztgtfgxhdh.chromatic.com)

🚀 CI was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.z6.web.core.windows.net/apim-451-conf-options-entrypoint/index.html)
_Notes_: The deployed app is linked to the management API of APIM master. (Same login and password as APIM master)

### CODE REVIEW

File: gravitee-apim-console-webui/src/management/api/creation/v4/api-creation-v4.component.spec.ts [hunk 1]
```diff
@@ -434,6 +502,20 @@ describe('ApiCreationV4Component', () => {
     expectEntrypointsGetRequest(entrypoints);
 
     await step2Harness.fillAndValidate(entrypoints.map((entrypoint) => entrypoint.id));
+
+    await fillStepTwoOneAndValidate(entrypoints);
+  }
+
+  async function fillStepTwoOneAndValidate(
```
Comment (ThibaudAV) [inline gravitee-apim-console-webui/src/management/api/creation/v4/api-creation-v4.component.spec.ts]:
```suggestion
  async function fillAndValidateStep2Entrypoints2Config(
```

Comment (gcusnieux) [reply gravitee-apim-console-webui/src/management/api/creation/v4/api-creation-v4.component.spec.ts]: ✅

File: gravitee-apim-console-webui/src/management/api/creation/v4/services/api-creation-stepper.service.ts [hunk 2]
```diff
@@ -94,10 +95,14 @@ export class ApiCreationStepperService {
    * Add a secondary step after the current step.
    */
   public addSecondaryStep(step: NewSecondaryApiCreationStep) {
+    if (this.steps.find(({ id }) => id === step.id)) {
```
Comment (ThibaudAV) [inline gravitee-apim-console-webui/src/management/api/creation/v4/services/api-creation-stepper.service.ts:98]: 🤔 I wonder if we could just calculate the generateStepId id before and check if it exists

File: gravitee-apim-console-webui/src/management/api/creation/v4/steps/api-creation-steps-common.component.scss [hunk 3]
```diff
@@ -47,4 +47,11 @@ $typography: map.get(gio.$mat-theme, typography);
     font-weight: 700;
     vertical-align: bottom;
   }
+
+  &__form-container {
+    border: 1px solid #d3d5dc;
```
Comment (ThibaudAV) [inline gravitee-apim-console-webui/src/management/api/creation/v4/steps/api-creation-steps-common.component.scss]: Cloud we use a ui-p conventional color on this ?

File: gravitee-apim-console-webui/src/management/api/creation/v4/steps/step-2-entrypoints/step-2-entrypoints-2-config.component.ts [hunk 4]
```diff
@@ -30,6 +32,10 @@ export class Step2Entrypoints2ConfigComponent implements OnInit, OnDestroy {
   private unsubscribe$: Subject<void> = new Subject<void>();
 
   public formGroup: FormGroup;
+  public selectedEntrypoints: { id: string; name: string }[];
+  public entrypointSchemas: Record<string, GioJsonSchema>;
+  public entrypointInitialValues: Record<string, any>;
+  public entrypointFormGroups: Record<string, FormGroup>;
```
Comment (ThibaudAV) [inline gravitee-apim-console-webui/src/management/api/creation/v4/steps/step-2-entrypoints/step-2-entrypoints-2-config.component.ts:38]: WDYT to try to replace this by `formGroup.get(entrypoint.id)` directly in template ?

Comment (ThibaudAV) [reply gravitee-apim-console-webui/src/management/api/creation/v4/steps/step-2-entrypoints/step-2-entrypoints-2-config.component.ts:38]: and try to have something like this 
```
this.formGroup = currentStepPayload.selectedEntrypoints.reduce...
```
🤷‍♂️ ?

File: gravitee-apim-console-webui/src/management/api/creation/v4/steps/step-2-menu-item/step-2-menu-item.component.scss [hunk 5]
```diff
@@ -0,0 +1,23 @@
+@use '@angular/material' as mat;
+@use '@gravitee/ui-particles-angular' as gio;
+
+.step-2-menu-item {
+  display: flex;
+  flex-direction: column;
+  font-weight: bold;
+  padding: 16px;
+  position: relative;
```
Comment (ThibaudAV) [inline gravitee-apim-console-webui/src/management/api/creation/v4/steps/step-2-menu-item/step-2-menu-item.component.scss:9]: I wonder if it is necessary to have Pos: relative and Pos absolute ? maybe we can juste have a padding for line no ?

**Review by ThibaudAV — APPROVED** [review]
i've see some bug with stepper. I check it and fix in another PR

File: gravitee-apim-console-webui/src/management/api/creation/v4/services/api-creation-stepper.service.ts (hunk 2 above)
Comment (gcusnieux) [reply gravitee-apim-console-webui/src/management/api/creation/v4/services/api-creation-stepper.service.ts:98]: The problem with the fn generateStepId, is that we increment according to the label

File: gravitee-apim-console-webui/src/management/api/creation/v4/steps/step-2-menu-item/step-2-menu-item.component.scss (hunk 5 above)
Comment (gcusnieux) [reply gravitee-apim-console-webui/src/management/api/creation/v4/steps/step-2-menu-item/step-2-menu-item.component.scss:9]: I don't think, but we can try

File: gravitee-apim-console-webui/src/management/api/creation/v4/services/api-creation-stepper.service.ts (hunk 2 above)
Comment (ThibaudAV) [reply gravitee-apim-console-webui/src/management/api/creation/v4/services/api-creation-stepper.service.ts:98]: haha right

File: gravitee-apim-console-webui/src/management/api/creation/v4/steps/step-2-menu-item/step-2-menu-item.component.scss (hunk 5 above)
Comment (ThibaudAV) [reply gravitee-apim-console-webui/src/management/api/creation/v4/steps/step-2-menu-item/step-2-menu-item.component.scss:9]: as you wish I was just wondering. if not 🚀

**Pull request merged**